Lib/defconQt/glyphView.py

```diff
@@ -187,27 +292,38 @@ def __init__(self, glyph, parent=None):
         glyphMenu.addAction("&Jump", self.changeGlyph, "J")
         menuBar.addMenu(glyphMenu)
 
-        toolBar = QToolBar(self)
+
+        self._displaySettings = DisplayStyleSettings("&Display", self, self._displayChanged)
+        menuBar.addMenu(self._displaySettings.menuWidget)
+
+        self._toolBar = toolBar = QToolBar(self)
         toolBar.setMovable(False)
         toolBar.setContentsMargins(2, 0, 2, 0)
-        selectionToolButton = toolBar.addAction("Selection", self.view.setSceneSelection)
+        selectionToolButton = toolBar.addAction("Selection", self._redirect('view', 'setSceneSelection'))
         selectionToolButton.setCheckable(True)
         selectionToolButton.setChecked(True)
         selectionToolButton.setIcon(QIcon(":/resources/cursor.svg"))
-        penToolButton = toolBar.addAction("Pen", self.view.setSceneDrawing)
+        penToolButton = toolBar.addAction("Pen", self._redirect('view', 'setSceneDrawing'))
         penToolButton.setCheckable(True)
-        penToolButton.setIcon(QIcon(":/resources/curve.svg"))
-        rulerToolButton = toolBar.addAction("Ruler", self.view.setSceneRuler)
+        selectionToolButton.setIcon(QIcon(":/resources/cursor.svg"))
+        rulerToolButton = toolBar.addAction("Ruler", self._redirect('view', 'setSceneRuler'))
         rulerToolButton.setCheckable(True)
-        rulerToolButton.setIcon(QIcon(":/resources/ruler.svg"))
-        knifeToolButton = toolBar.addAction("Knife", self.view.setSceneKnife)
+        selectionToolButton.setIcon(QIcon(":/resources/cursor.svg"))
+        knifeToolButton = toolBar.addAction("Knife", self._redirect('view', 'setSceneKnife'))
         knifeToolButton.setCheckable(True)
-        knifeToolButton.setIcon(QIcon(":/resources/cut.svg"))
+        selectionToolButton.setIcon(QIcon(":/resources/cursor.svg"))
         # http://www.setnode.com/blog/right-aligning-a-button-in-a-qtoolbar/
         spacer = QWidget()
         spacer.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding)
         toolBar.addWidget(spacer)
-        toolBar.addWidget(self.view.currentLayerBox())
+        self._currentLayerBox = QComboBox(self)
```

Comment:
So, the active layer is kept in the window when glyph is changed (the GlyphView is deleted and replaced by a new one)

> Is the goal to have all the external UI elements in the MainWindow rather than in the View? I've found handling that in the View more convenient in practise.

I wouldn't call it the goal. I rather wanted to create a kind of a hierarchy. The layers as they are implemented in ufo/defcon come with this strange concept, that each layer is in a way a font by it's own. To reflect that structure in ufo defcon has: font->layerSet->layers->glyphs and not, as one would intuitively think: font->glyphs->layers

I think it's good not to have a too big gap between the model (defcon) hierarchy and the view (UI) hierarchy. There's kind of a "natural" place for things to happen. Some things happen in the glyph, and some happen one level above.